### PR TITLE
Move `propOr` to `getPropOr`

### PR DESCRIPTION
**Pull request by evilsoft**
## Let's Move Out!!
image

This PR addressed the `propOr` portion of [this issue](https://github.com/evilsoft/crocks/issues/329) and deprecates the `propOr` function in favor of `getPropOr` to match the convention we are moving to. 

This also updates all the docs to reflect this change.

### CODE REVIEW

**Comment by evilsoft** [comment]
> I didn't spot a changelog. I rely pretty heavily on these, so do make sure to communicate the change clearly.

Do you mean a change log in the PR?

**Comment by bennypowers** [comment]
> > I didn't spot a changelog. I rely pretty heavily on these, so do make sure to communicate the change clearly.
> 
> Do you mean a change log in the PR?

Right. I imagine that will go out with all the rest of the changes, though.

**Comment by evilsoft** [comment]
@dalefrancis88 @bennypowers or @karthikiyengar 
This is all up-to-date and 🔴 👁 for review if anyone wants to grab it!

File: docs/src/pages/docs/functions/helpers.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 title: "Helpers"
 description: "Helper functions"
 layout: "notopic"
-functions: ["assign", "assoc", "binary", "compose", "composek", "composep", "composes", "curry", "defaultprops", "defaultto", "dissoc", "frompairs", "lifta2", "lifta3", "liftn", "mapprops", "mapreduce", "mconcat", "mconcatmap", "mreduce", "mreducemap", "nary", "objof", "omit", "once", "partial", "pick", "pipe", "pipek", "pipep", "pipes", "propor", "proppathor", "setpath", "setprop", "tap", "unary", "unit", "unsetpath", "unsetprop"]
+functions: ["assign", "assoc", "binary", "compose", "composek", "composep", "composes", "curry", "defaultprops", "defaultto", "dissoc", "frompairs", "getpropor", "lifta2", "lifta3", "liftn", "mapprops", "mapreduce", "mconcat", "mconcatmap", "mreduce", "mreducemap", "nary", "objof", "omit", "once", "partial", "pick", "pipe", "pipek", "pipep", "pipes", "propor", "proppathor", "setpath", "setprop", "tap", "unary", "unit", "unsetpath", "unsetprop"]
 weight: 20
 ---
```
Comment (dalefrancis88) [inline docs/src/pages/docs/functions/helpers.md]: We are leading 'propOr' in there for a little while like the others we've made redundant?

File: docs/src/pages/docs/functions/helpers.md [hunk 2]
```diff
@@ -340,6 +340,52 @@ you provide an `undefined` values for the second, that `Pair` will not be
...
+getPropOr :: a -> (String | Integer) -> b -> a
+```
+
+If you want some safety around pulling a value out of an Object or Array with a
+single key or index, you can always reach for `getPropOr`. Well, as long as you
+are working with non-nested data that is. Just tell `getPropOr` either the key
+or index you are interested in, and you will get back a function that will take
+anything and return the wrapped value if the key/index is defined. If the
+key/index is not defined however, you will get back the provided default value.
+
```
Comment (dalefrancis88) [inline docs/src/pages/docs/functions/helpers.md]: I know I didn't mention it on the others, but do we want to reference that this used to be called prop or is replacing prop?

File: src/helpers/getPropOr.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
+/** @license ISC License (c) copyright 2019 original and current authors */
+/** @author Ian Hofmann-Hicks */
+
+const curry = require('../core/curry')
+const isDefined = require('../core/isDefined')
+const isEmpty = require('../core/isEmpty')
+const isNil = require('../core/isNil')
+const isInteger = require('../core/isInteger')
```
Comment (dalefrancis88) [inline src/helpers/getPropOr.js]: Does the order matter with imports in code or only in docs?

Comment (evilsoft) [reply src/helpers/getPropOr.js]: It matters everywhere! great 👁 will fix.

File: docs/src/pages/docs/functions/helpers.md (hunk 2 above)
Comment (evilsoft) [reply docs/src/pages/docs/functions/helpers.md]: Oh. I like that, we will have to be removed when we release 1.0, but I am okay with that.

Will add. Should I update the other functions as well on this PR or would you think that would be too much noise?

File: docs/src/pages/docs/functions/helpers.md (hunk 1 above)
Comment (evilsoft) [reply docs/src/pages/docs/functions/helpers.md]: yep. Do you think we shouldn't?

File: docs/src/pages/docs/functions/helpers.md (hunk 2 above)
Comment (dalefrancis88) [reply docs/src/pages/docs/functions/helpers.md]: My preference is to be as specific as possible with PR's, that being said up to you, i think the change is small enough to slip in as part of this

File: docs/src/pages/docs/functions/helpers.md (hunk 1 above)
Comment (dalefrancis88) [reply docs/src/pages/docs/functions/helpers.md]: No just double checking, at the time i was doing this on my phone so checking the code elsewhere would have been a pain 😅

**Comment by evilsoft** [comment]
@dalefrancis88 okay updated those combinator examples. You did not miss them before, they came in on a rebase.

**Comment by evilsoft** [comment]
image

**Pull request merged**